### PR TITLE
fix: fix miscellaneous linting problems in the codebase

### LINKED ISSUE #700: fix: fix miscellaneous linting problem in the codebase
**Describe the bug**

- [x] golint: https://goreportcard.com/report/github.com/Azure/aad-pod-identity#golint
- [x] misspell: https://goreportcard.com/report/github.com/Azure/aad-pod-identity#misspell
- [x] turn on misspell linting tool in CI

**Steps To Reproduce**

**Expected behavior**

**AAD Pod Identity version**

**Kubernetes version**

**Additional context**

### PR DESCRIPTION
**Pull request by chewong**
**Reason for Change**:

fixes #700 

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [ ] ran `make precommit`

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: pkg/apis/aadpodidentity/types.go [hunk 1]
```diff
@@ -5,39 +5,67 @@ import (
...
 const (
-	CRDGroup    = "aadpodidentity.k8s.io"
-	CRDVersion  = "v1"
+	// CRDGroup is the group name of aad-pod-identity CRDs.
+	CRDGroup = "aadpodidentity.k8s.io"
+
+	// CRDVersion is the version of the CRD group.
+	CRDVersion = "v1"
+
+	// CRDLabelKey is the
```
Comment (aramase) [inline pkg/apis/aadpodidentity/types.go:50]: nit: `CRDLabelKey is the static label that's used in pod`

File: pkg/apis/aadpodidentity/types.go [hunk 2]
```diff
@@ -5,39 +5,67 @@ import (
...
 	CRDLabelKey = "aadpodidbinding"
 
+	// BehaviorKey is the key that describes the behavior of aad-pod-identity.
```
Comment (aramase) [inline pkg/apis/aadpodidentity/types.go:53]: Can we add - Supported values: `namespaced`, used for running in namespaced mode. AzureIdentity, AzureIdentityBinding and pod in the same namespace will only be matched for this behavior

File: pkg/apis/aadpodidentity/types.go [hunk 3]
```diff
@@ -146,40 +180,41 @@ type AzureIdentitySpec struct {
...
 type AssignedIDState int
 
 const (
-	//Created - Default state of the assigned identity
+	// Created - Default state of the assigned identity
 	Created AssignedIDState = 0
-	//Assigned - When the underlying platform assignment of EMSI is complete
-	//the state moves to assigned
+
+	// Assigned - When the underlying platform assignment of EMSI is complete,
```
Comment (aramase) [inline pkg/apis/aadpodidentity/types.go:196]: nit: EMSI/managed identity

File: pkg/apis/aadpodidentity/types.go [hunk 4]
```diff
@@ -146,40 +180,41 @@ type AzureIdentitySpec struct {
...
+	AzureIDResource = "azureidentities"
+
+	// AzureIDBindingResource is the name of AzureIdentityBinding.
+	AzureIDBindingResource = "azureidentitybindings"
+
+	// AzureAssignedIDResource is the name of AzureAssignedIdentity.
+	AzureAssignedIDResource = "azureassignedidentities"
+
+	// AzureIdentityExceptionResource is the name of AzureIdentityException.
 	AzureIdentityExceptionResource = "azurepodidentityexceptions"
```
Comment (aramase) [inline pkg/apis/aadpodidentity/types.go:212]: Let's change this var to `AzurePodIdentityExceptionResource` to align with the other resource variables

File: pkg/apis/aadpodidentity/utils.go [hunk 5]
```diff
@@ -1,5 +1,6 @@
 package aadpodidentity
 
+// IsNamespacedIdentity returns true if azureID is an namespaced identity.
```
Comment (aramase) [inline pkg/apis/aadpodidentity/utils.go:3]: nit: s/an/a

File: pkg/crd/crd.go [hunk 6]
```diff
@@ -59,7 +59,7 @@ type ClientInt interface {
 	ListPodIdentityExceptions(ns string) (res *[]aadpodid.AzurePodIdentityException, err error)
 }
 
-// NewCRDClientLite ...
+// NewCRDClientLite returns
```
Comment (aramase) [inline pkg/crd/crd.go:62]: nit: returns a new crd lite client

File: pkg/crd/crd.go [hunk 7]
```diff
@@ -370,6 +371,7 @@ func (c *Client) Upgrade(resource string, i runtime.Object) (map[string]runtime.
 	return m, nil
 }
 
+// UpgradeAll performs type upgrade to for all aad-pod-identity CRD.
```
Comment (aramase) [inline pkg/crd/crd.go:374]: nit: s/CRD/CRDs

File: pkg/crd/crd.go [hunk 8]
```diff
@@ -444,7 +446,7 @@ func (c *Client) StartLite(exit <-chan struct{}) {
 	klog.Info("CRD lite informers started ")
 }
 
-// Start ...
+// Start starts various go routines to watch for any CRD-related changes.
```
Comment (aramase) [inline pkg/crd/crd.go:449]: nit: `starts all informer routines to watch for CRD-related changes`

File: pkg/metrics/metrics.go [hunk 9]
```diff
@@ -30,31 +30,43 @@ const (
...
+	// AdalTokenOperationName represents the duration of obtaining a token.
 	AdalTokenOperationName = "adal_token"
-	// GetVmssOperationName ...
+
+	// GetVmssOperationName represents the duration of a GET request to a VMSS instance.
 	GetVmssOperationName = "vmss_get"
-	// PutVmssOperationName ...
+
+	// PutVmssOperationName represents the duration of a PUT request to a VMSS instance.
 	PutVmssOperationName = "vmss_create_or_update"
```
Comment (aramase) [inline pkg/metrics/metrics.go:46]: Now that we have switched to using `PATCH`, I think we should rename this to `PatchVMSSOperationName = "vmss_update"`

File: pkg/stats/stats.go [hunk 10]
```diff
@@ -7,94 +7,143 @@ import (
...
+	ExceptionList Type = "Pod Identity Exception listing"
+
+	// AssignedIDList represents the duration it takes to list AzureAssignedIdentities.
+	AssignedIDList Type = "Assigned ID listing"
+
+	// CloudGet represents the duration it takes to complete a GET request to ARM in a given sync cycle.
+	CloudGet Type = "Cloud provider get"
+
+	// CloudPut represents the duration it takes to complete a PUT request to ARM in a given sync cycle.
+	CloudPut Type = "Cloud provider put"
```
Comment (aramase) [inline pkg/stats/stats.go:55]: we should change this to `CloudUpdate Type = "Cloud provider update"`

File: pkg/metrics/metrics.go (hunk 9 above)
Comment (chewong) [reply pkg/metrics/metrics.go:46]: Changed to `UpdateVMSSOperationName = "vmss_update"` to unify variables name between metrics.go and stats.go.

**Pull request merged**